### PR TITLE
[pvr] change to new addon to kodi callback way

### PR DESCRIPTION
**Pull request by AlwinEsch**
## Description
This is the first major change to become multiple instances to work.

There are the PVR callbacks moved to the CPVRClient class. Reason is
to have this parts together and later with multiple instances to call
always right related CPVRClient class.

On addon is currently nothing needed to change, only a recompile to
match them.

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/addons/PVRClient.cpp [hunk 1]
```diff
@@ -669,8 +700,8 @@ PVR_ERROR CPVRClient::GetEPGForChannel(const CPVRChannelPtr &channel, CPVREpg *e
   ADDON_HANDLE_STRUCT handle;
   handle.callerAddress  = this;
   handle.dataAddress    = epg;
-  handle.dataIdentifier = bSaveInDb ? 1 : 0; // used by the callback method CAddonCallbacksPVR::PVRTransferEpgEntry()
-  PVR_ERROR retVal = m_struct.GetEpg(&handle,
+  handle.dataIdentifier = bSaveInDb ? 1 : 0; // used by the callback method CPVRClient::PVRTransferEpgEntry()
```
Comment (ksooo) [inline xbmc/addons/PVRClient.cpp]: The method mentioned in the comment imo should read <code>CPVRClient::cb_transfer_epg_entry</code>

File: xbmc/addons/PVRClient.cpp [hunk 2]
```diff
@@ -1565,29 +1589,432 @@ void CPVRClient::OnSystemSleep(void)
...
+    msg = strMessage;
+
+  CServiceBroker::GetPVRManager().ConnectionStateChange(client, std::string(strConnectionString), newState, msg);
+}
+
+typedef struct EpgEventStateChange
+{
+  int             iClientId;
+  unsigned int    iUniqueChannelId;
+  CPVREpgInfoTagPtr  event;
```
Comment (ksooo) [inline xbmc/addons/PVRClient.cpp]: Could you please fix the spaces? I suggest to put only one blank between type and variable name.

**Review by ksooo — APPROVED** [review]
Looking good. Please don't forget to open PRs to bump API version and addon version and for changelog update for all PVR addons. Keep on rocking!

**Comment by AlwinEsch** [comment]
jenkins build this please

**Comment by AlwinEsch** [comment]
jenkins build this please

**Pull request merged**